build/NuGetPackages/CreateNuGetPackages.proj

```diff
@@ -58,5 +58,32 @@
 
     <Message Importance="High" Text="@(NuSpecs->'%(Filename)') NuGet Package -> $(PackagesOutDir)\@(NuSpecs->'%(Filename)').$(NuGetPackageVersion).nupkg" />
   </Target>
+
+<Target Name="GetNuGetPackageVersionEx" DependsOnTargets="GetNuGetPackageVersion">
+  
+  <PropertyGroup>
+    <_BuildRevisionRegEx>\d{8}\.(?&lt;Revision&gt;\d{1,2})</_BuildRevisionRegEx>
+  </PropertyGroup>
   
+  <!-- Display an error if the build number from VSTS is not in the format yyyyMMdd-r -->
+  <Error Text="The BUILD_NUMBER property '$(BUILD_NUMBER)' is not in the expected format.  Expected the format of '$(_BuildRevisionRegEx)'." Condition="'$(BUILD_NUMBER)' != '' And !$([System.Text.RegularExpressions.Regex]::IsMatch($(BUILD_NUMBER), $(_BuildRevisionRegEx)))" />
+  
+  <!--
+    This target is used to customize the versions generated by NerdBank.GitVersioning.  The $(PrereleaseVersion) is read from version.json and
+    then the date and build number is appended.
+  -->
+  <PropertyGroup>
+    <BuildNumber>0</BuildNumber>
+    <!-- Local builds always have a revision of 00 -->
+    <BuildRevision Condition="'$(BUILD_NUMBER)' == ''">00</BuildRevision>
+    
+    <!-- Official builds have a revision starting with 01 every day -->
+    <BuildRevision Condition="'$(BUILD_NUMBER)' != ''">$([System.Int32]::Parse($([System.Text.RegularExpressions.Regex]::Match($(BUILD_NUMBER), $(_BuildRevisionRegEx)).Groups["Revision"].Value)).ToString('D2'))</BuildRevision>
```

Comment:
Should we just change the build number format to be `$(rev:.rr)` and avoid this regex extraction (at the cost of requiring build configuration)?